src/processBadResponse.js

```diff
@@ -1,33 +1,52 @@
+import { defaultErrorCodes } from './constants';
 import { setAuthError } from './okapiActions';
 
-function getDefaultError() {
-  return {
-    code: 'default.error',
+const getDefaultError = () => (
+  {
+    code: defaultErrorCodes.DEFAULT_ERROR,
     type: 'error',
-  };
-}
+  }
+);
+
+const getDefaultServerError = () => (
+  {
+    code: defaultErrorCodes.DEFAULT_SERVER_ERROR,
+    type: 'error'
+  }
+);
 
-function getProcessedErrors(errorMessage, response) {
+const getProcessedErrors = (errorMessage, response) => {
   const processedErrors = [];
   const { status } = response;
 
   if (status === 422) {
     try {
-      const { errors } = JSON.parse(errorMessage);
-
-      processedErrors.push(...errors);
+      if (typeof errorMessage === 'object') {
+        const { errors } = errorMessage;
+        processedErrors.push(...errors);
+      } else {
+        const { errors } = JSON.parse(errorMessage);
+        processedErrors.push(...errors);
+      }
     } catch (e) {
       processedErrors.push(getDefaultError());
     }
+  } else if (status >= 500) {
+    processedErrors.push(getDefaultServerError());
   } else {
     processedErrors.push(getDefaultError());
   }
 
   return processedErrors;
-}
+};
 
 export default function processBadResponse(store, response) {
```

Comment:
I would suggest to rewrite it to async/await. Also please pass only dispatch instead of store.